### ISSUE TITLE
webrtc/pc/peerconnectionfactory.h file not found.

### ISSUE THREAD

**Issue by dkushner** [sourcey/libsourcey#170]
After working all day to get this library to finally compile, I gave up trying to get LibSourcey to recognize my built version of WebRTC and just downloaded the pre-compiled libs. I finally have it half-way compiling but am now getting this:

```
/Users/me/CLionProjects/heave/cmake-build-debug/libsourcey/source/src/webrtc/include/scy/webrtc/peerfactorycontext.h:18:10: fatal error: 'webrtc/pc/peerconnectionfactory.h' file not found
#include "webrtc/pc/peerconnectionfactory.h"
```

It is the only error, and I have no idea why this specific header is not found when I have specified WEBRTC_ROOT correctly and that file clearly exists in the include directory under that path.

**Comment by auscaster** [comment]
There are often versioning issues with WebRTC since the libraries change so
fast. You can try to remove the header if it is no longer a part of WebRTC?

**Comment by dkushner** [comment]
That's the kicker. The file it is complaining about actually _does_ exist in the WebRTC root that I gave CMake. 

![file list](http://i.imgur.com/NnhJftU.png)

**Comment by auscaster** [comment]
OK thats very strange indeed, I've never had any issue like this. What does
your directory structure look like with webrtc and libsourcey, also what
does your cmake command line look like?

**Comment by dkushner** [comment]
Layout looks like ![this](http://i.imgur.com/xxyPA91.png) where libsourcey is included via CMake's ExternalProject:

```
externalproject_add(
        libsourcey
        DEPENDS webrtc

        GIT_REPOSITORY http://github.com/sourcey/libsourcey

        PREFIX ${CMAKE_BINARY_DIR}/libsourcey
        BINARY_DIR ${CMAKE_BINARY_DIR}/libsourcey/build
        SOURCE_DIR ${CMAKE_BINARY_DIR}/libsourcey/source
        DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/libsourcey/source
        STAMP_DIR ${CMAKE_BINARY_DIR}/libsourcey/stamp
        TMP_DIR ${CMAKE_BINARY_DIR}/libsourcey/tmp

        CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                   -DWEBRTC_ROOT_DIR=${webrtc_ROOT_DIR}
                   -DCMAKE_PROJECT_LibSourcey_INCLUDE=${CMAKE_MODULE_PATH}/FixWebRTC.cmake
                   -DBUILD_SHARED_LIBS=OFF -DBUILD_APPLICATIONS=OFF
                   -DBUILD_MODULES=OFF -DBUILD_SAMPLES=off
                   -DBUILD_TESTS=OFF -DWITH_FFMPEG=ON
                   -DWITH_OPENCV=OFF -DWITH_WEBRTC=ON
                   -DWITH_WEBRTC=ON -DBUILD_MODULE_archo=ON
                   -DBUILD_MODULE_base=ON -DBUILD_MODULE_crypto=ON
                   -DBUILD_MODULE_http=ON -DBUILD_MODULE_json=ON
                   -DBUILD_MODULE_av=ON -DBUILD_MODULE_av=ON
                   -DBUILD_MODULE_net=ON -DBUILD_MODULE_net=ON
                   -DBUILD_MODULE_socketio=ON -DBUILD_MODULE_stun=ON
                   -DBUILD_MODULE_webrtc=ON -DBUILD_MODULE_turn=ON
                   -DBUILD_MODULE_util=ON
)

`webrtc_ROOT_DIR` is set to the folder shown in the previous screenshot and `FixWebRTC.cmake` is a necessary hack to include a CMake file that defines `-DWEBRTC_POSIX=1` because this caused issues earlier in the build process.

**Comment by auscaster** [comment]
Did you make any progress here?

The fact that you are using `ExternalProject` must be the issue. Somehow WebRTC directories are not getting included.

Internally those directories are stored in `WEBRTC_INCLUDE_DIRS`. Perhaps you could debug there?

**Comment by hankbao** [comment]
I've encountered this issue recently. After checked `CXX.includecache` and other `.cmake` files for the webrtc module, I found out the problem.

It turns out there are several webrtc-include directories been set in the CXX_INCLUDES flag. It's basically caused by changing WEBRTC_ROOT_DIR flag for cmake during make. 

At first I tried to build libsourcey with webrtcbuilds at master branch. However the source code structure of webrtc has changed so libsourcey fails to build. Then I switched to webrtc branch-heads/59 and removed artifacts built previously. Thus the issue surfaced.

Just removed the build directory and went though the build process again. Everything is fine now.
